Source/AwsCommonRuntimeKit/http/HTTPStreamCallbackCore.swift

```diff
@@ -94,8 +94,13 @@ private func onResponseBody(stream: UnsafeMutablePointer<aws_http_stream>?,
 
     let incomingBodyFn = httpStreamCbData.requestOptions.onIncomingBody
     let callbackBytes = Data(bytesNoCopy: bufPtr, count: bufLen, deallocator: .none)
-    incomingBodyFn(callbackBytes)
-    return AWS_OP_SUCCESS
+
+    do {
+        try incomingBodyFn(callbackBytes)
+        return AWS_OP_SUCCESS
+    } catch {
+        return AWS_OP_ERR
```

Comment:
We should raise an error, something like `return aws_raise_error(Int32(AWS_ERROR_HTTP_CALLBACK_FAILURE.rawValue))`

Reply:
We should also make other callbacks throwable apart from onComplete.